yt_dlp/extractor/roblox.py

```diff
@@ -0,0 +1,129 @@
+        toolbox_asset_data = toolbox_result.get('asset') or {}
+        toolbox_creator_data = toolbox_result.get('creator') or {}
+        toolbox_audio_data = toolbox_asset_data.get('audioDetails') or {}
+
+        info_dict = {
+            'id': asset_id,
+            'title': toolbox_asset_data.get('name') or item_container_attrs.get('data-item-name'),
+            'uploader': toolbox_creator_data.get('name') or asset_uploader_name,
+            'uploader_id': str_or_none(toolbox_creator_data.get('id')) or asset_uploader_id,
+            # TODO: Sound effects have separate kinds of categories 
+            'categories': toolbox_asset_data.get('assetGenres') or [strip_or_none(get_element_by_class('item-genre', webpage))],
+            'like_count': str_to_int(extract_attributes(get_element_html_by_class('favoriteCount', webpage)).get('title')),
+            'timestamp': unified_timestamp(toolbox_asset_data.get('createdUtc')),
+            'modified_timestamp': unified_timestamp(toolbox_asset_data.get('updatedUtc')),  # TODO: Extract from webpage
+            'track': toolbox_audio_data.get('title'),
+            'artist': toolbox_audio_data.get('artist'),
+            'genre': try_call(lambda: toolbox_audio_data['musicGenre'].capitalize())
+        }
+
+        cdn_result = self._download_json(
+            f'https://assetdelivery.roblox.com/v1/assetId/{asset_id}',
+            asset_id, note='Downloading file data JSON', headers={
+                'Accept': 'application/json',
+                'roblox-browser-asset-request': 'true'
+            })
+        asset_file_url = cdn_result.get('location')
+        if not asset_file_url:
+            if asset_type == 'Audio':
+                media_play_icon_div = get_element_html_by_class('MediaPlayerIcon')
+                if not media_play_icon_div:
+                    self.raise_no_formats('This audio is unavailable', expected=True, video_id=asset_id)
+                asset_file_url = self._search_regex(r'data-mediathumb-url=["\']https?://[^"\']+["\']', media_play_icon_div, 'audio preview URL')
+            elif is_logged_out:  # assetdelivery API randomly requires auth cookies
+                self.raise_login_required(metadata_available=True)
+            else:
+                self.raise_no_formats(
+                    traverse_obj(cdn_result, ('errors', ..., 'message'), default='Unable to fetch asset', expected_type=str, get_all=False),
+                    video_id=asset_id)
+
+        if asset_file_url:
+            # Assets have no file extension and use binary/octet-stream as Content-Type
+            pp = FFmpegPostProcessor(self._downloader)
+            self.to_screen(f'{asset_id}: Checking file format with ffprobe')
+            try:
+                metadata = pp.get_metadata_object(asset_file_url)
+            except PostProcessingError as err:
+                raise ExtractorError(err.msg, expected=True)
```

Comment:
There's actually a feature request to support this in core code https://github.com/yt-dlp/yt-dlp/issues/613

That aside, is there no way to avoid this in the extractor?

Reply:
The most important metadata are setting acode/vcodec='none' and ext. I assume codec can be handled with `asset_type`

Reply:
You can only narrow down the formats, e.g. audio can only be uploaded as ogg/vorbis or mp3, and while video is most likely webm/vp9 (since video can only be uploaded by select ppl, we can't even be sure about this), the metadata hints at mp4 muxing being possible